deployment/provision_vagrant_vm.sh

```diff
@@ -93,6 +93,27 @@ cp $REPO_FOLDER/deployment/manage_autocompletion.sh /etc/bash_completion.d/
 # install chrome, see: https://github.com/puppeteer/puppeteer/issues/7740
 apt-get -q install -y chromium-browser
 
+# install firefox and geckodriver
```

Comment:
Can you elaborate a bit further? I don't know anything about debian packaging, what is the config below doing?

Reply:
Since January, Mozilla provides their own PPA. We should be able to get firefox up using just:
```
# https://support.mozilla.org/en-US/kb/install-firefox-linux#w_install-firefox-deb-package-for-debian-based-distributions
sudo install -d -m 0755 /etc/apt/keyrings
wget -q https://packages.mozilla.org/apt/repo-signing-key.gpg -O- | sudo tee /etc/apt/keyrings/packages.mozilla.org.asc > /dev/null
echo "deb [signed-by=/etc/apt/keyrings/packages.mozilla.org.asc] https://packages.mozilla.org/apt mozilla main" | sudo tee -a /etc/apt/sources.list.d/mozilla.list > /dev/null
echo '
Package: *
Pin: origin packages.mozilla.org
Pin-Priority: 1000
' | sudo tee /etc/apt/preferences.d/mozilla 
sudo apt-get update && sudo apt-get install firefox
```

I don't know if their repo also contains a matching geckodriver package though. I would hope it does.

(The current code here still has the geckodriver vs firefox version mismatch problem, right?)

Reply:
Well no, geckodriver is not jet a part of it. I found https://bugzilla.mozilla.org/show_bug.cgi?id=1876704 tracking it

Reply:
From the linked bug report:
> For us it will be important that we are not bound to Firefox release cycles. Compared to Chrome and chromedriver, which require a version match for each release, geckodriver can be used with a variety of Firefox releases at maximum back to the latest ESR release. If there are issues that need to be fixed and released quickly we shouldn't be blocked on getting beta/release approval to uplift patches.

So, for firefox and geckodriver, there is no such problem.